src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexPrefixAnalyzer.cs

```diff
@@ -12,8 +11,89 @@ namespace System.Text.RegularExpressions
     /// <summary>Detects various forms of prefixes in the regular expression that can help FindFirstChars optimize its search.</summary>
     internal static class RegexPrefixAnalyzer
     {
+        // The minimum number of use to match that it would make sense to use a MultiStringMatcher.
+        private const int MinMultiPrefixes = 5;
+
+        /// <summary>
+        /// Tries to create a trie from the leading fixed part of a <see cref="RegexNode"/>.
+        /// If using the trie is estimated not to be profitable, a leading substring will be
+        /// attempted to be found, and the trie will be <see langword="null"/>.
+        /// </summary>
+        internal static (string? SinglePrefix, List<TrieNode>? PrefixTrie) CreatePrefixMatcher(RegexNode node)
+        {
+            if ((node.Options & (RegexOptions.RightToLeft | RegexOptions.NonBacktracking | RegexOptions.Compiled)) != 0)
```

Comment:
I'm confused now.  Why do the source generator and RegexCompiler have code in support of the prefix matcher if we disable this for RegexOptions.Compiled?

Reply:
I added this relatively late after seeing disappointing results for the compiled mode. I should fine-tune it more and re-enable it. Perhaps directly emitting the AC logic would improve things.